library/camel-kamelets-utils/src/main/java/org/apache/camel/kamelets/utils/serialization/InflightAvroSchemaResolver.java

```diff
@@ -0,0 +1,21 @@
+package org.apache.camel.kamelets.utils.serialization;
+
+import com.fasterxml.jackson.core.FormatSchema;
+import com.fasterxml.jackson.dataformat.avro.AvroSchema;
+
+import org.apache.avro.Schema;
+import org.apache.camel.Exchange;
+import org.apache.camel.component.jackson.SchemaResolver;
+
+public class InflightAvroSchemaResolver implements SchemaResolver {
+
+    @Override
+    public FormatSchema resolve(Exchange exchange) {
+        String schemaJson = (String) exchange.getProperty("schema");
```

Comment:
Same as the previous PR, should these exchange properties be removed so they dont leak out of the kamelet